src/cloud/benchflow/sender/sender.go

```diff
@@ -379,27 +431,35 @@ func main() {
 	kafkaIp = viper.GetString("kafka_host")
 	kafkaPort = viper.GetString("kafka_port")
 	cassandraHost = viper.GetString("cassandra_host")
+	driverMemory = viper.GetString("driver_memory")
+	executorMemory = viper.GetString("executor_memory")
 	minioHost = viper.GetString("minio_host")
-	minio_port = viper.GetString("minio_port")
+	minioPort = viper.GetString("minio_port")
+	minioAccessKey = viper.GetString("minio_access_key")
+	minioSecretKey = viper.GetString("minio_secret_key")
+	runsBucket = viper.GetString("runs_bucket")
 	sparkHome = viper.GetString("spark_home")
 	sparkMaster = viper.GetString("spark_master")
-	spark_port = viper.GetString("spark_port")
+	sparkPort = viper.GetString("spark_port")
 	alluxio_port = viper.GetString("alluxio_port")
 	pysparkCassandraVersion = viper.GetString("pyspark_cassandra_version")
 	analysersPath = viper.GetString("analysers_path")
 	transformersPath = viper.GetString("transformers_path")
 	configurationsPath = viper.GetString("configurations_path")
+	// TODO: Take from the right file on Minio
```

Comment:
An ID uniquely identifies the entity to which it relates. The experiment number is not an ID by itself, but just a number. The only way to uniquely identify an experiment is by combining the userID, the experiment name, and the experiment number.

Example: `simone.fooExp.1`.

As far as collectors are concerned, I inject the following environment variables in the collectors descriptors:

- `SUT_NAME`: the name of the S.U.T.
- `SUT_VERSION`: the version of the S.U.T.
- `BENCHFLOW_EXPERIMENT_ID`: as defined above
- `BENCHFLOW_TRIAL_ID`: ID for a trial, i.e., experimentID + trial number (e.g., `simone.fooExp.1.1`)
- `BENCHFLOW_TRIAL_TOTAL`: the total number of trials

If you need different or more data to be passed to the collectors, I can modify the implementation accordingly. Retrieving the YAML file without knowing the experiment ID is impossible, because the file itself is identified by that ID. How can you retrieve the definition of an experiment, if you don't know what experiment you want?